Dockerfile

```diff
@@ -3,17 +3,29 @@ RUN apt-get update && \
     apt-get install -y software-properties-common && \
     add-apt-repository -y ppa:cwchien/gradle
 RUN apt-get update && \
+    apt install --reinstall -y software-properties-common && \
+    add-apt-repository ppa:openjdk-r/ppa && \
+    apt-get update && \
     apt-get install -y git \
                        ssh \
                        tar \
                        gzip \
                        ca-certificates \
                        openjdk-8-jdk \
-                       gradle \
                        python2.7 \
                        python-pip \
                        zlib1g-dev
+RUN apt-get install -y openjdk-11-jdk && \
+    apt-get purge -y openjdk-8-jdk
```

Comment:
What's the purpose of installing then purging later?

Reply:
Oh, its this deeky issue with java11 and gradle. If you dont do that, Gradle doesnt work correctly. Can't remember the exact issue